runtime/hydradx/src/benchmarking/dca.rs

```diff
@@ -0,0 +1,340 @@
+
+fn schedule_fake(
+	owner: AccountId,
+	asset_in: AssetId,
+	asset_out: AssetId,
+	amount: Balance,
+) -> Schedule<AccountId, AssetId, BlockNumber> {
+	let schedule1: Schedule<AccountId, AssetId, BlockNumber> = Schedule {
+		owner,
+		period: 3u32,
+		total_amount: 1100 * ONE,
+		max_retries: None,
+		stability_threshold: None,
+		slippage: Some(Permill::from_percent(15)),
+		order: Order::Buy {
+			asset_in,
+			asset_out,
+			amount_out: amount,
+			max_amount_in: Balance::MAX,
+			route: create_bounded_vec(vec![Trade {
+				pool: PoolType::Omnipool,
+				asset_in,
+				asset_out,
+			}]),
+		},
+	};
+	schedule1
+}
+
+fn get_named_reseve_balance(token_id: AssetId, seller: AccountId) -> Balance {
+	Currencies::reserved_balance_named(&NamedReserveId::get(), token_id, &seller)
+}
+
+fn schedule_buy_fake(
+	owner: AccountId,
+	asset_in: AssetId,
+	asset_out: AssetId,
+	amount: Balance,
+) -> Schedule<AccountId, AssetId, BlockNumber> {
+	let schedule1: Schedule<AccountId, AssetId, BlockNumber> = Schedule {
+		owner,
+		period: 3u32,
+		total_amount: 2000 * ONE,
+		max_retries: None,
+		stability_threshold: None,
+		slippage: Some(Permill::from_percent(15)),
+		order: Order::Buy {
+			asset_in,
+			asset_out,
+			amount_out: amount,
+			max_amount_in: Balance::MAX,
+			route: create_bounded_vec(vec![Trade {
+				pool: PoolType::Omnipool,
+				asset_in,
+				asset_out,
+			}]),
+		},
+	};
+	schedule1
+}
+
+fn schedule_sell_fake(
+	owner: AccountId,
+	asset_in: AssetId,
+	asset_out: AssetId,
+	amount: Balance,
+) -> Schedule<AccountId, AssetId, BlockNumber> {
+	let schedule1: Schedule<AccountId, AssetId, BlockNumber> = Schedule {
+		owner,
+		period: 3u32,
+		total_amount: 2000 * ONE,
+		max_retries: None,
+		stability_threshold: None,
+		slippage: Some(Permill::from_percent(100)),
+		order: Order::Sell {
+			asset_in,
+			asset_out,
+			amount_in: amount,
+			min_amount_out: Balance::MIN,
+			route: create_bounded_vec(vec![Trade {
+				pool: PoolType::Omnipool,
+				asset_in,
+				asset_out,
+			}]),
+		},
+	};
+	schedule1
+}
+
+fn set_period(to: u32) {
+	while System::block_number() < Into::<BlockNumber>::into(to) {
+		let b = System::block_number();
+
+		System::on_finalize(b);
+		EmaOracle::on_finalize(b);
+
+		System::on_initialize(b + 1_u32);
+		EmaOracle::on_initialize(b + 1_u32);
+
+		System::set_block_number(b + 1_u32);
+	}
+}
+
+pub fn create_bounded_vec(trades: Vec<Trade<AssetId>>) -> BoundedVec<Trade<AssetId>, ConstU32<5>> {
+	let bounded_vec: BoundedVec<Trade<AssetId>, ConstU32<5>> = trades.try_into().unwrap();
+	bounded_vec
+}
+
+fn create_account_with_native_balance() -> Result<AccountId, DispatchError> {
+	let caller: AccountId = account("provider", 1, 1);
+	let token_amount = 200 * ONE;
+	<Currencies as MultiCurrencyExtended<AccountId>>::update_balance(0u32, &caller, token_amount as i128)?;
+
+	Ok(caller)
+}
+
+runtime_benchmarks! {
+	{Runtime, pallet_dca}
+
+	on_initialize_with_buy_trade{
```

Comment:
there is only an omnipool trade involved within this. The weight of this will be used to explicitly calculate the DCA execution fee in the DCA pallet. Don't we need to incorporate here more trade types such as stablepool?

Reply:
Trade type used in the benchmarks doesn't matter because we use dummy router that does nothing.